src/Microsoft.DotNet.Interactive/KernelExtensions.cs

```diff
@@ -424,5 +424,22 @@ public static T LogEventsToPocketLogger<T>(this T kernel)
 
             await VisitSubkernelsAsync(kernel, onVisit, recursive);
         }
+
+        public static IEnumerable<Kernel> SubkernelsAndSelf(
```

Comment:
I'd prefer to avoid the duplicate logic here. The `Visit*` methods can probably just call this now, no?

Reply:
makes sense